src/anomalib/metrics/evaluator.py

```diff
@@ -0,0 +1,131 @@
+"""Evaluator module for LightningModule."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from collections.abc import Sequence
+from typing import Any
+
+from lightning.pytorch import Callback, LightningModule, Trainer
+from lightning.pytorch.utilities.types import STEP_OUTPUT
+from torch import nn
+from torch.nn import ModuleList
+from torchmetrics import Metric
+
+from anomalib.metrics import AnomalibMetric
+
+
+class Evaluator(nn.Module, Callback):
+    """Evaluator module for LightningModule.
+
+    The Evaluator module is a PyTorch module that computes and logs metrics during
+    validation and test steps. Each AnomalyModule should have an Evaluator module as
+    a submodule to compute and log metrics during validation and test steps. An Evaluation
+    module can be passed to the AnomalyModule as a parameter during initialization. When
+    no Evaluator module is provided, the AnomalyModule will use a default Evaluator module
+    that logs a default set of metrics.
+
+    Args:
+        val_metrics (Sequence[AnomalibMetric], optional): Validation metrics.
+            Defaults to ``[]``.
+        test_metrics (Sequence[AnomalibMetric], optional): Test metrics.
+            Defaults to ``[]``.
+        compute_on_cpu (bool, optional): Whether to compute metrics on CPU.
+            Defaults to ``True``.
+
+    Examples:
+        >>> from anomalib.metrics import F1Score, AUROC
+        >>> from anomalib.data import ImageBatch
+        >>> import torch
+        >>>
+        >>> f1_score = F1Score(fields=["pred_label", "gt_label"])
+        >>> auroc = AUROC(fields=["pred_score", "gt_label"])
+        >>>
+        >>> evaluator = Evaluator(test_metrics=[f1_score])
+    """
+
+    def __init__(
+        self,
+        val_metrics: AnomalibMetric | Sequence[AnomalibMetric] | None = None,
+        test_metrics: AnomalibMetric | Sequence[AnomalibMetric] | None = None,
+        compute_on_cpu: bool = True,
+        **kwargs,
```

Comment:
what would be kwargs?

Reply:
kwargs are not needed here. I'll remove them